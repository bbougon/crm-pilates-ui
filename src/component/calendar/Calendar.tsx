import '@zach.codes/react-calendar/dist/calendar-tailwind.css';
import * as React from "react";
import {MainContainer} from "../../const/containers";
import {Grid} from "@material-ui/core";
import {PilatesMonthlyCalendar} from "./PilatesMonthlyCalendar";

export class Calendar extends React.Component<{ date: Date }> {

    render() {
        return (
            <MainContainer>
                <Grid container>
                    <Grid item xs={11}>
                        <PilatesMonthlyCalendar date={this.props.date || new Date()}/>
                    </Grid>
                </Grid>
            </MainContainer>
        )
    }
}

export default Calendar;