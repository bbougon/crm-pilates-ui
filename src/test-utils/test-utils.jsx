import React from 'react'
import {act, render as rtlRender} from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import clientsReducer from "../features/clientsSlice"
import sessionsReducer from "../features/sessionsSlice"
import classroomsReducer from "../features/classroomSlice"
// Import your own reducer

const reducer = {
    sessions: sessionsReducer,
    clients: clientsReducer,
    classrooms: classroomsReducer
}

function render(
    ui,
    {
        preloadedState,
        store = configureStore({ reducer: reducer, preloadedState }),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export async function actThenSleep(ms) {
    await act(() => sleep(20));
}

export function sleep(ms) {
    return new Promise((resolve) => setTimeout(resolve, ms));
}

export * from '@testing-library/react'
export { render }