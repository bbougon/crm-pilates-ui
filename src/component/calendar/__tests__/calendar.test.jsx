import {render} from "../../../test-utils/test-utils";
import {screen, waitFor} from "@testing-library/react";
import React from "react";
import Calendar from "../Calendar";
import {
    APIErrorBody,
    RequestHandlerBuilders,
    ServerBuilder,
    SessionXLinkValueHeaderBuilder,
    XLinkHeaderBuilder
} from "../../../test-utils/server/server";

describe('Calendar page', () => {
    const emptyClients = new RequestHandlerBuilders().get("/clients").ok().body([]).build();

    const server = new ServerBuilder().serve(emptyClients)

    beforeAll(() => server.listen())

    afterEach(() => server.resetHandlers())

    afterAll(() => server.close())


    it('should have a plus button on each day', async () => {
        server.resetHandlers(emptyClients, new RequestHandlerBuilders().get("/sessions")
            .ok()
            .body([])
            .header(new XLinkHeaderBuilder().value(new SessionXLinkValueHeaderBuilder().current(new Date())))
            .build()
        )
        render(<Calendar date={new Date("2021-10-10T11:20:00")}/>)

        expect(await waitFor(() =>screen.getAllByTestId("AddIcon"))).toHaveLength(26)
    })


    describe('Load sessions facing an error', () => {

        it('should display the error', async () => {
            server.resetHandlers(
                new RequestHandlerBuilders()
                    .get("/sessions")
                    .unprocessableEntity()
                    .body(new APIErrorBody()
                        .dummyDetail()
                        .build())
                    .build(),
                emptyClients
            )

            render(<Calendar date={new Date("2021-10-10T11:20:00")}/>)

            expect(await screen.findByText("An error occurred (see message below):", {selector: 'h5'})).toBeInTheDocument()
            expect(await screen.findByText("an error message", {selector: 'p'})).toBeInTheDocument()
            expect(await screen.findByText("an error type", {selector: 'p'})).toBeInTheDocument()
        })
    })


})