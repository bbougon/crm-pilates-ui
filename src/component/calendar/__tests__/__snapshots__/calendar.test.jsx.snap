// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar page should be displayed 1`] = `
"[36m<div>[39m
  [36m<div[39m
    [33mclass[39m=[32m\\"sc-bdfBQB gOWVrL\\"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"MuiGrid-root MuiGrid-container\\"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11\\"[39m
      [36m>[39m
        [36m<div[39m
          [33mclass[39m=[32m\\"rc-flex rc-justify-end rc-mb-4\\"[39m
        [36m>[39m
          [36m<button[39m
            [33mclass[39m=[32m\\"rc-cursor-pointer\\"[39m
          [36m>[39m
            [0mPrevious[0m
          [36m</button>[39m
          [36m<div[39m
            [33maria-label[39m=[32m\\"Current Month\\"[39m
            [33mclass[39m=[32m\\"rc-ml-4 rc-mr-4 rc-w-32 rc-text-center\\"[39m
          [36m>[39m
            [0mSeptember[0m
          [36m</div>[39m
          [36m<button[39m
            [33mclass[39m=[32m\\"rc-cursor-pointer\\"[39m
          [36m>[39m
            [0mNext[0m
          [36m</button>[39m
        [36m</div>[39m
        [36m<div[39m
          [33mclass[39m=[32m\\"rc-bg-white rc-border-l-2 rc-border-t-2\\"[39m
        [36m>[39m
          [36m<div[39m
            [33mclass[39m=[32m\\"rc-grid rc-grid-cols-1 sm:rc-grid-cols-2 md:rc-grid-cols-4 lg:rc-grid-cols-7\\"[39m
          [36m>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mSunday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mMonday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mTuesday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mWednesday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mThursday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mFriday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Day of Week\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m>[39m
              [0mSaturday[0m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Empty Day\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m/>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Empty Day\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m/>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Empty Day\\"[39m
              [33mclass[39m=[32m\\"rc-border-b-2 rc-p-2 rc-border-r-2 lg:rc-block rc-hidden\\"[39m
            [36m/>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Events for day 1\\"[39m
              [33mclass[39m=[32m\\"rc-h-48 rc-p-2 rc-border-b-2 rc-border-r-2\\"[39m
            [36m>[39m
              [36m<div[39m
                [33mclass[39m=[32m\\"rc-flex rc-justify-between\\"[39m
              [36m>[39m
                [36m<div[39m
                  [33mclass[39m=[32m\\"rc-font-bold\\"[39m
                [36m>[39m
                  [0m1[0m
                [36m</div>[39m
                [36m<div[39m
                  [33mclass[39m=[32m\\"lg:rc-hidden rc-block\\"[39m
                [36m>[39m
                  [0mWednesday[0m
                [36m</div>[39m
              [36m</div>[39m
              [36m<ul[39m
                [33mclass[39m=[32m\\"rc-divide-gray-200 rc-divide-y rc-overflow-hidden rc-max-h-36 rc-overflow-y-auto\\"[39m
              [36m>[39m
                [36m<div[39m
                  [33mclass[39m=[32m\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root\\"[39m
                [36m>[39m
                  [36m<div[39m
                    [33mclass[39m=[32m\\"MuiBox-root css-1bvc4cc\\"[39m
                  [36m>[39m
                    [36m<button[39m
                      [33mclass[39m=[32m\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root\\"[39m
                      [33mtabindex[39m=[32m\\"0\\"[39m
                      [33mtype[39m=[32m\\"button\\"[39m
                    [36m>[39m
                      [36m<svg[39m
                        [33maria-hidden[39m=[32m\\"true\\"[39m
                        [33mclass[39m=[32m\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root\\"[39m
                        [33mdata-testid[39m=[32m\\"AddIcon\\"[39m
                        [33mfocusable[39m=[32m\\"false\\"[39m
                        [33mviewBox[39m=[32m\\"0 0 24 24\\"[39m
                      [36m>[39m
                        [36m<path[39m
                          [33md[39m=[32m\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"[39m
                        [36m/>[39m
                      [36m</svg>[39m
                      [36m<span[39m
                        [33mclass[39m=[32m\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"[39m
                      [36m/>[39m
                    [36m</button>[39m
                  [36m</div>[39m
                [36m</div>[39m
              [36m</ul>[39m
            [36m</div>[39m
            [36m<div[39m
              [33maria-label[39m=[32m\\"Events for day 2\\"[39m
              [33mclass[39m=[32m\\"rc-h-48 rc-p-2 rc-border-b-2 rc-border-r-2\\"[39m
            [36m>[39m
              [36m<div[39m
                [33mclass[39m=[32m\\"rc-flex rc-justify-between\\"[39m
              [36m>[39m
                [36m<div[39m
                  [33mclass[39m=[32m\\"rc-font-bold\\"[39m
                [36m>[39m
                  [0m2[0m
                [36m</div>[39m
                [36m<div[39m
                  [33mclass[39m=[32m\\"lg:rc-hidden rc-block\\"[39m
               ..."
`;
